### PR TITLE
cleared terms and fees error message

### PR DESCRIPTION
**Pull request by adeel0202**
### Details
Cleared the error message when both of the required checkboxes are checked on `Terms and fees` page

### Fixed Issues
$ #8244

### Tests
- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

1. Go to Settings
2. Go to Payments
3. Tap on `Transfer Balance` (you must have some balance available)
4. On `Additional details` step, fill in all the details and press `Save & continue`
5. On `Onfido` step, verify your identity and press `continue`
6. On `Terms and fees` step, press `Enable payments` button without checking the required checkboxes to show the error message
7. Now check both of the required checkboxes
8. Observe that the error message is cleared when both required checkboxes are checked

#### Web

https://user-images.githubusercontent.com/13783842/160076342-ea60628e-2a0b-4e23-a9a4-d0f7a78d2b81.mov

#### Mobile Web

https://user-images.githubusercontent.com/13783842/160076388-a5797573-a319-46d4-bf41-00704af4fbc5.mp4

#### Desktop

https://user-images.githubusercontent.com/13783842/160076498-0165e63a-5bdb-407b-bb6c-106c52e1d3d1.mov

#### iOS

https://user-images.githubusercontent.com/13783842/160076524-443fe499-8eb0-40dd-be5b-c1548008bfcc.mp4

#### Android

https://user-images.githubusercontent.com/13783842/160076544-97b61e6a-c8a9-4e7e-a0c0-ba93d99e54e6.mov

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@adeel0202 
QA Steps are not clear,
Can you add the steps 
How to test on Local, Staging/Production Environments.

**Comment by adeel0202** [comment]
@Santhosh-Sellavel, I don't know the right steps to reach that page since I was never able to add a test bank account, so I followed [these instructions](https://github.com/Expensify/App/issues/7838#issuecomment-1059282757) for testing purposes.

**Comment by Santhosh-Sellavel** [comment]
I think we need the steps for how to test it in a staging or production environment too. So @adeel0202 Kindly post in slack and ask for some help on this one.

**Comment by adeel0202** [comment]
@Santhosh-Sellavel, I have updated the QA steps.

File: src/pages/EnablePayments/TermsStep.js [hunk 1]
```diff
@@ -46,13 +46,22 @@ class TermsStep extends React.Component {
...
     togglePrivacyPolicy() {
         this.setState(prevState => ({
             hasAcceptedPrivacyPolicyAndWalletAgreement: !prevState.hasAcceptedPrivacyPolicyAndWalletAgreement,
-        }));
+        }), () => this.clearError());
+    }
+
+    clearError() {
+        if (!this.state.hasAcceptedDisclosure
+            || !this.state.hasAcceptedPrivacyPolicyAndWalletAgreement) {
```
Comment (Santhosh-Sellavel) [inline src/pages/EnablePayments/TermsStep.js:60]: This could be single line.

File: src/pages/EnablePayments/TermsStep.js [hunk 2]
```diff
@@ -46,13 +46,22 @@ class TermsStep extends React.Component {
...
     }
 
     togglePrivacyPolicy() {
         this.setState(prevState => ({
             hasAcceptedPrivacyPolicyAndWalletAgreement: !prevState.hasAcceptedPrivacyPolicyAndWalletAgreement,
-        }));
+        }), () => this.clearError());
+    }
+
+    clearError() {
```
Comment (Santhosh-Sellavel) [inline src/pages/EnablePayments/TermsStep.js:58]: The method could be moved above `toggleDisclosure`

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]

**Comment by adeel0202** [comment]
PR updated.

**Comment by Santhosh-Sellavel** [comment]
Thanks, @adeel0202!

Avoid too many commits for simple changes like this one.
It is always good to run lint tests locally before committing.

**Review by stitesExpensify — APPROVED** [review]
LGTM!

**Comment by Santhosh-Sellavel** [comment]
Sorry, @stitesExpensify I forget to approve & add the PR Reviewer Checklist here. Everything LGTM me as well!

**Comment by mvtglobally** [comment]
@neil-marcellini @stitesExpensify Can you help QA this one internally? This is again about Transfer Balance. We are not able to QA

**Comment by stitesExpensify** [comment]
IMO that means we should just check it off on the deploy checklist for now, do you agree @neil-marcellini ?

**Comment by neil-marcellini** [comment]
Yeah it's probably fine to let it through given that it's a small change.

**Comment by stitesExpensify** [comment]
Cool. @mvtglobally feel free to check this off!

**Pull request merged**